src/swarm-engine/Swarm/Game/State.hs

```diff
@@ -245,9 +246,10 @@ robotsInArea (Cosmic subworldName o) d rs = mapMaybe (rm IM.!?) rids
   rm = rs ^. robotMap
   rl = rs ^. robotsByLocation
   rids =
-    concatMap IS.elems $
-      getElemsInArea o d $
-        M.findWithDefault mempty subworldName rl
+    concatMap IS.elems
+      . getElemsInArea o d
+      . MM.toMap
```

Comment:
This is a zero-cost coercion. See:

https://github.com/jonathanknowles/monoidmap/blob/f54db7f51a5bed6b4213674be96a4e87475a78c6/components/monoidmap-internal/Data/MonoidMap/Internal.hs#L574-L575